build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -855,60 +878,63 @@ class _SingleBuild {
     for (final output in outputs) {
       final node = _assetGraph.get(output)!;
       if (node.type == NodeType.generated &&
-          node.generatedNodeState.wasOutput) {
+          node.generatedNodeState!.wasOutput) {
         await _delete(output);
       }
     }
   }
 
-  Future<void> _buildGlobNode(AssetNode globNode) async {
-    if (globNode.globNodeState.pendingBuildAction == PendingBuildAction.none) {
+  Future<void> _buildGlobNode(AssetId id) async {
+    if (_assetGraph.get(id)!.globNodeState!.pendingBuildAction ==
+        PendingBuildAction.none) {
       return;
     }
 
-    return _lazyGlobs.putIfAbsent(globNode.id, () async {
-      var potentialNodes =
+    return _lazyGlobs.putIfAbsent(id, () async {
+      final globNodeConfiguration = _assetGraph.get(id)!.globNodeConfiguration!;
+      var potentialIds =
           _assetGraph
-              .packageNodes(globNode.id.package)
+              .packageNodes(id.package)
               .where((n) => n.isFile && n.isTrackedInput)
               .where(
                 (n) =>
                     n.type != NodeType.generated ||
-                    n.generatedNodeConfiguration.phaseNumber <
-                        globNode.globNodeConfiguration.phaseNumber,
-              )
-              .where(
-                (n) => globNode.globNodeConfiguration.glob.matches(n.id.path),
+                    n.generatedNodeConfiguration!.phaseNumber <
+                        globNodeConfiguration.phaseNumber,
               )
+              .where((n) => globNodeConfiguration.glob.matches(n.id.path))
+              .map((node) => node.id)
               .toList();
 
-      for (final node in potentialNodes) {
+      for (final potentialId in potentialIds) {
+        final node = _assetGraph.get(potentialId)!;
```

Comment:
will this node be different from the node we extracted the id from above? (aka I don't understand why we map it to the id, then get the node based on id?)

Reply:
It was to avoid having locals around that would become stale after the `_buildAsset` in the next loop.

This whole method can be improved--added comments and some for loops instead of where / where / map, hopefully this looks better :) thanks.